worker/caasoperator/caasoperator.go

```diff
@@ -214,6 +216,35 @@ func (op *caasOperator) loop() (err error) {
 		)
 	}
 
+	var (
+		watcher   *remotestate.RemoteStateWatcher
+		watcherMu sync.Mutex
+	)
+
+	restartWatcher := func() error {
+		watcherMu.Lock()
+		defer watcherMu.Unlock()
+
+		if watcher != nil {
+			// watcher added to catacomb, will kill uniter if there's an error.
+			worker.Stop(watcher)
```

Comment:
Could we please not swallow an error coming from Stop()? Even logging it might help in the long run...

Reply:
we're not consistent here (across the code base). this was cribbed as is from uniter